motionplan/rrt.go

```diff
@@ -270,16 +259,41 @@ func sumCosts(path []node) float64 {
 	return cost
 }
 
-func transformNodes(path []node, transformBy spatialmath.Pose) []node {
-	transformedNodes := []node{}
-	for _, n := range path {
-		newNode := &basicNode{
-			q:      n.Q(),
-			cost:   n.Cost(),
-			pose:   spatialmath.Compose(n.Pose(), transformBy),
-			corner: n.Corner(),
+type rrtPlan struct {
+	traj Trajectory
+	path Path
+
+	// nodes corresponding to inputs can be cached with the Plan for easy conversion back into a form usable by RRT
+	// depending on how the trajectory is constructed these may be nil and should be computed before usage
+	nodes []node
+}
+
+func newRRTPlan(solution []node, sf *solverFrame, relative bool) (*rrtPlan, error) {
+	if len(solution) < 2 {
+		return nil, errors.New("cannot construct a Plan using fewer than two nodes")
+	}
+	traj := sf.nodesToTrajectory(solution)
+	path, err := newRelativePath(solution, sf)
+	if err != nil {
+		return nil, err
+	}
+	if relative {
+		path, err = newAbsolutePathFromRelative(path)
```

Comment:
Based on the name of `newAbsolutePathFromRelative I would expect it to return an absolute path. I don't understand why we would do that if `relative` is `true`.

Reply:
This was confusing though. I updated the names